### ISSUE TITLE
no warning message about factor to character coercion shown when binding dataframes

### ISSUE THREAD

**Issue by salim-b** [hadley/dplyr#2317]
With [dplyr v0.4.0](https://github.com/hadley/dplyr/blob/master/NEWS.md#minor-improvements) warning messages were introduced when a coercion from type factor to character is needed to produce a valid result using `bind_rows()` and other operations.

If I run the following, the warning message is shown as expected:
```R
> bind_rows(data_frame(a = factor("a")), data_frame(a = "a"))
# A tibble: 2 × 1
      a
  <chr>
1     a
2     a
Warnmeldung:
In bind_rows_(x, .id) :
  binding factor and character vector, coercing into character vector
```
Now if I switch the order of the dataframes, no warning message appears although the same coercion from type `factor` to `character` is applied to the second dataframe:
```R
> bind_rows(data_frame(a = "a"), data_frame(a = factor("a")))
# A tibble: 2 × 1
      a
  <chr>
1     a
2     a
```

This behaviour seems confusing and in my eyes the warning message should appear in both cases. 

I'm using `dplyr 0.5.0` with `R 3.3.2` on `Ubuntu Linux 16.04 LTS`.

**Comment by krlmlr** [comment]
Thanks, confirmed.

**Comment by krlmlr** [comment]
Warn in the second case, too

**Comment by hadley** [comment]
@zeehio I'd love your thoughts here too, if you have the time.

**Comment by zeehio** [comment]
@hadley Quick fix written in PR #2478
